packages/nimble-components/src/table/models/table-validator.ts

```diff
@@ -0,0 +1,58 @@
+import type { TableRecord, TableValidity } from '../types';
+
+/**
+ * Helper class for the nimble-table to validate that the table's configuration
+ * is valid and report which aspects of the configuration are valid or invalid.
+ */
+export class TableValidator<TData extends TableRecord> {
+    private duplicateRowId = false;
+    private missingRowId = false;
+    private invalidRowId = false;
+
+    public getValidity(): TableValidity {
+        return {
+            duplicateRowId: this.duplicateRowId,
+            missingRowId: this.missingRowId,
+            invalidRowId: this.invalidRowId
+        };
+    }
+
+    public isValid(): boolean {
+        return Object.values(this.getValidity()).every(x => x === false);
+    }
+
+    public validateDataIds(
+        data: TData[],
+        idFieldName: string | null | undefined
+    ): boolean {
+        // Start off by assuming all IDs are valid.
+        this.duplicateRowId = false;
+        this.missingRowId = false;
+        this.invalidRowId = false;
+
+        if (idFieldName == null) {
+            return true;
+        }
+
+        const ids = new Set<string>();
+        for (const record of data) {
+            if (!Object.prototype.hasOwnProperty.call(record, idFieldName)) {
```

Comment:
nit: [Object.hasOwn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) is a shortcut for exactly this

Reply:
I get this error when I try that:
`TS2550: Property 'hasOwn' does not exist on type 'ObjectConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.`

Reply:
eh no biggie for now, it's just syntax sugar, can make a concerted effort to pickup some new es2022 syntax later